docs/src/components/code-block.astro

```diff
@@ -0,0 +1,21 @@
+---
+import { Code } from "astro:components";
+
+import { CopyButton } from "@/components/copy-button";
+
+interface Props {
+  code: string;
+  lang?: "bash" | "js";
+  copy?: boolean;
+}
+
+const { code, lang = "js", copy = false } = Astro.props;
+---
+
+<div class="relative">
+  <Code code={code} lang={lang} theme="poimandres" />
+
+  <div class="absolute right-2 top-1/2 -translate-y-1/2">
+    {copy && <CopyButton value={code} client:load />}
+  </div>
+</div>
```

Comment:
This component will wrap **all** our code examples in the main "docs" page with 2 divs unnecessarily for the single use-case where we need the copy button.

Let's remove the whole abstraction.

This cuts out about 15% of the size of the docs.html file.
```suggestion
```